plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcColumnHandle.java

```diff
@@ -211,14 +219,21 @@ public Builder setComment(Optional<String> comment)
             return this;
         }
 
+        public Builder setOriginalColumnName(String originalColumnName) {
+            this.originalColumnName = originalColumnName;
+            return this;
+        }
+
         public JdbcColumnHandle build()
         {
-            return new JdbcColumnHandle(
+            JdbcColumnHandle handle =  new JdbcColumnHandle(
                     columnName,
                     jdbcTypeHandle,
                     columnType,
                     nullable,
                     comment);
+            handle.originalColumnName = originalColumnName;
```

Comment:
`originalColumnName` is not added as it should be as `final` field as this creates a waterfall of changes in other places in the code that don't use the builder, but a direct constructor call. Since this solution proposal might be dumped and started from scratch after the review, I didn't want to invest that time yet.